### PR TITLE
fix : Added tests for curriculum order

### LINKED ISSUE #43407: Add tests for curriculum order
[This PR](https://github.com/freeCodeCamp/freeCodeCamp/pull/42182#discussion_r704011339) added a new block in responsive web design with the same order number as another and all the tests passed. The result was that the lessons jumped between the two blocks after finishing each lesson.

We could possibly add some tests to ensure that the superBlock order numbers are unique, the block order numbers are unique within their superBlock, and the challenge order numbers are unique within their block.

If anything else comes to mind, feel free to share or add it - perhaps a warning if an order number is skipped or doesn't start at 1? I believe the [block numbers for responsive web design](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/_meta/basic-html-and-html5/meta.json#L5) start at zero, they start at two in [front end development libraries](https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/_meta/bootstrap/meta.json#L5), and the rest start at one. Maybe a warning/log for something like that.

### PR DESCRIPTION
**Pull request by aksamota**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #43407

**Added tests for curriculum order.** 
- Checks for unique superBlock and superOrder
- Checks for unique order numbers for each block in a superBlock
- Console Warning if the order numbers do not start at 1 or are not contiguous
- Checks for unique challenge IDs inside each block

Also modified the order numbers and superOrder for legacy certification blocks.

### CODE REVIEW

File: curriculum/test/test-curriculum-order.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    it('order numbers are expected to be contiguos for each superBlock', function () {
+      Object.keys(superBlocks).forEach(key => {
+        const orders = Object.keys(superBlocks[key]).map(toNumber);
+        orders.sort((a, b) => a < b);
+
+        // expect(orders).to.deep.equal([...Array(orders.length).keys()]);
+        for (let i = 1; i < orders.length; i++) {
+          if (orders[i] !== orders[i - 1] + 1) {
+            console.warn(`orders for ${key} are not contiguos`);
```
Comment (nhcarrigan) [inline curriculum/test/test-curriculum-order.js:106]:
```suggestion
    it('order numbers are expected to be contiguous for each superBlock', function () {
      Object.keys(superBlocks).forEach(key => {
        const orders = Object.keys(superBlocks[key]).map(toNumber);
        orders.sort((a, b) => a < b);

        // expect(orders).to.deep.equal([...Array(orders.length).keys()]);
        for (let i = 1; i < orders.length; i++) {
          if (orders[i] !== orders[i - 1] + 1) {
            console.warn(`orders for ${key} are not contiguous`);
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Hey there,

Thanks for getting this started - a quick nitpick on the spelling here.

Additionally, it looks like this test timed out on the GitHub action (which caps at 6 hours of runtime). Can you confirm that this test does not take 6 hours locally?

**Comment by ankitkumarsamota121** [comment]
Hi, I can confirm that this test does not take 6 hours locally. But I have observed that the complete test suite does not complete in the specified time. I ran the test suite just after cloning the repo and all the tests were passing in a few minutes. The only changes I have made are the addition of the tests for curriculum order (`curriculum/test/test-curriculum-order.js`) and changes of order and superOrder in the legacy challenges. 

I am unable to figure out what is causing the issue. Are there any other tests that depend on the order and superOrder in metafiles?

**Comment by ojeytonwilliams** [comment]
As far as I can tell it's an issue with how our curriculum tests are structured.  Because we pass the `--delay` flag to `mocha` we need to invoke the `run()` command once all the tests are ready to go.  Since run is invoked by `test-challenges.js` I suspect there's a race condition between the two test files.

I 'solved' it locally by copying the new tests into `populateTestsForLang`, which makes sure all the `describe`s get called before `run`, which isn't terribly elegant, but seems to work.  The smart approach is probably to pass `--file` to `mocha` to specify `test-challenges.js` and rework `test-curriculum-order` to return a function analogous to `populateTestsForLang` which can be called before `run()`.

**Comment by raisedadead** [comment]
Hi, 

We are closing old issues/pull requests that we think are stale. If you need still need help with this or are fixing this, please open a fresh issue/pull request.

Thanks for your understanding. 

Happy contributing.

**Pull request closed without merging**